sdk/cosmos/azure-cosmos/azure/cosmos/aio/_container.py

```diff
@@ -301,6 +304,8 @@ async def read_item(
         if max_integrated_cache_staleness_in_ms is not None:
             validate_cache_staleness_value(max_integrated_cache_staleness_in_ms)
             request_options["maxIntegratedCacheStaleness"] = max_integrated_cache_staleness_in_ms
+        if self.container_link in self.__get_client_container_caches():
```

Comment:
I wonder whether this logic can be moved into GetHeaders, seems that is a centralized places to populate headers

Reply:
It is currently set in GetHeaders, we are passing the value here as we do with partition key values into the request options which are passed into GetHeaders.